src/internal/connector/onedrive/restore.go

```diff
@@ -207,7 +214,17 @@ func RestoreCollection(
 					restoredIDs[trimmedName] = itemID
 
 					deets.Add(itemPath.String(), itemPath.ShortRef(), "", true, itemInfo)
+
+					// Mark it as success without processing .meta
+					// file if we are not restoring permissions
+					if !enablePermissionsRestore {
+						metrics.Successes++
+					}
 				} else if strings.HasSuffix(name, MetaFileSuffix) {
```

Comment:
Nit: this if-else chain is also a great place to break out code into small, testable functions.  The more handling we add to it (like permission restore control here), the more unwieldy it becomes.